src/Engine/IO/Window.cpp

```diff
@@ -150,34 +160,23 @@ void Window::update(float dt) {
         return;
     }
 
+    glfwPollEvents();
+
     /* Update mouse */
-    double x, y;
-    glfwGetCursorPos(window, &x, &y);
-    Mouse::update(x, y);
+    if (!cursorEnabled) {
+        double x, y;
+        glfwGetCursorPos(window, &x, &y);
+        Mouse::update(x, y);
+    }
 
 #ifdef DEBUG_MODE
     /* Update ImGui */
-    // TODO: clean up this code
-    static bool priorCursorState;
-    imGuiTimer += dt;
```

Comment:
Why take out the timer?

Reply:
no longer necessary. imgui is toggled on a key press

Reply:
Does it not bug out? Time timer was added because key presses are interpreted many times per second which causes the imgui panes to turn on and off repeatedly. The timer forces the panes to only be open or closed once per half second.

Reply:
nope. only happens on key down. so yes, imgui will open/close as quickly as you can press the key, but that's not a problem

Reply:
But that's why the timer was added in the first place

Reply:
no, it's only on the initial key down event, not on repeat events, and not on release. You can hold the key down and it's fine. If you want to mash the key, then yea, the menu's going to open every other time you press the key, but who cares?